lib/system/web/src/closure.rs

```diff
@@ -0,0 +1,60 @@
+use crate::prelude::*;
+
+use js_sys::Function;
+use wasm_bindgen::JsCast;
+use wasm_bindgen::convert::FromWasmAbi;
+
+
+
+// ======================
+// === ClosureStorage ===
+// ======================
+
+/// Constraint for JS closure argument types
+pub trait ClosureArg = FromWasmAbi + 'static;
+
+/// Function that can be wrapped into a `Closure`.
+pub trait ClosureFn<Arg> = FnMut(Arg) + 'static where Arg: ClosureArg;
+
+/// Stores an optional closure.
+/// The purpose it reduce boilerplate repeating when setting JS callbacks.
+#[derive(Debug,Derivative)]
+#[derivative(Default(bound=""))]
+pub struct ClosureStorage<Arg> {
+    /// The stored closure.
+    pub closure : Option<Closure<dyn FnMut(Arg)>>,
+}
```

Comment:
If this is only for storing optional closure, I would name it `OptionalClosure`. The name `ClosureStorage` does not indicate its real purpose. Moreover, this has a hardcoded type to be `FnMut`, whcile `Closure` is generic over the type. I would make this generic as well or name it `OptionalFnMutClosure`.

Reply:
done